internal/topology/check/compatibility_test.go

```diff
@@ -170,12 +174,220 @@ func TestCheckReferencedObjectsAreStrictlyCompatible(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			g := NewWithT(t)
 
-			err := ReferencedObjectsAreStrictlyCompatible(tt.current, tt.desired)
+			allErrs := ObjectsAreStrictlyCompatible(tt.current, tt.desired).ToAggregate()
 			if tt.wantErr {
-				g.Expect(err).To(HaveOccurred())
+				g.Expect(allErrs).To(HaveOccurred())
 				return
 			}
-			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(allErrs).ToNot(HaveOccurred())
+		})
+	}
+}
+
+func TestTemplatesAreCompatible(t *testing.T) {
```

Comment:
```suggestion
func TestLocalObjectTemplateAreCompatible(t *testing.T) {
```
nit, not blocking, can go in a follow up PR